src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetCoreMvc3Integration.cs

```diff
@@ -0,0 +1,495 @@
+            TargetMaximumVersion = MaximumVersion)]
+        public static void Rethrow_ExceptionContextSealed(object context, int opCode, int mdToken, long moduleVersionPtr)
+        {
+            Rethrow(context, opCode, mdToken, moduleVersionPtr, ExceptionContextSealedTypeName);
+        }
+
+        /// <summary>
+        /// Wrapper method used to catch unhandled exceptions in the incoming request pipeline for Microsoft.AspNetCore.Mvc.Core
+        /// </summary>
+        /// <param name="context">The DiagnosticSource that this extension method was called on.</param>
+        /// <param name="opCode">The OpCode used in the original method call.</param>
+        /// <param name="mdToken">The mdToken of the original method call.</param>
+        /// <param name="moduleVersionPtr">A pointer to the module version GUID.</param>
+        [InterceptMethod(
+            TargetAssembly = AspnetMvcCore,
+            TargetType = ResourceInvokerTypeName,
+            TargetSignatureTypes = new[] { ClrNames.Void, ResultExecutedContextSealedTypeName },
+            TargetMethod = nameof(Rethrow),
+            TargetMinimumVersion = MinimumVersion,
+            TargetMaximumVersion = MaximumVersion)]
+        public static void Rethrow_ResultExecutedContextSealed(object context, int opCode, int mdToken, long moduleVersionPtr)
+        {
+            Rethrow(context, opCode, mdToken, moduleVersionPtr, ResultExecutedContextSealedTypeName);
+        }
+
+        private static void Rethrow(object context, int opCode, int mdToken, long moduleVersionPtr, string contextType)
+        {
+            Action<object> instrumentedMethod;
+
+            try
+            {
+                var module = ModuleLookup.GetByPointer(moduleVersionPtr);
+                var concreteType = module.GetType(contextType);
+
+                instrumentedMethod =
+                    MethodBuilder<Action<object>>
+                       .Start(moduleVersionPtr, mdToken, opCode, nameof(Rethrow))
+                       .WithConcreteType(concreteType)
+                       .WithParameters(context)
+                       .WithNamespaceAndNameFilters(ClrNames.Void, contextType)
+                       .Build();
+            }
+            catch (Exception ex)
+            {
+                Log.ErrorRetrievingMethod(
+                    exception: ex,
+                    moduleVersionPointer: moduleVersionPtr,
+                    mdToken: mdToken,
+                    opCode: opCode,
+                    instrumentedType: ResourceInvokerTypeName,
+                    methodName: nameof(Rethrow),
+                    instanceType: null,
+                    relevantArguments: new[] { context.GetType().AssemblyQualifiedName });
+                throw;
+            }
+
+            AspNetCoreMvcContext integration = null;
+
+            try
+            {
+                if (context.TryGetPropertyValue("HttpContext", out object httpContext))
```

Comment:
We have a `GetProperty()` wrapper around `TryGetPropertyValue()` that removes the need for `if` and `out` parameters and allows chaining. I always meant to go back and remove `TryGetPropertyValue()` entirely, since it's not used anywhere (or maybe only in the older integrations?).

```csharp
var contextItems = context.GetProperty("HttpContext")
                          .GetProperty("Items")
                          .GetValueOrDefault() as IDictionary<object, object>;
```

No need to block on this right now, though.

Reply:
After writing this code sample I noticed you were using `GetProperty()` in `GetTagValues()` below. 😝

Reply:
Haha

Reply:
I actually like the if statements sometimes.